### PR TITLE
Rework recipe to support cmake

### LINKED ISSUE #15: Cmake config is missing
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

The libdwarf feedstock currently doesn’t support cmake usage of libdwarf which is problematic for anything trying to find_package(libdwarf). This is something now supported much better upstream so it should be possible to support in the feedstock.

### Installed packages

```shell
N/a
```

### Environment info

```shell
N/a
```

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
This PR updates the recipe to use the cmake build process which allows the library to be better used by cmake while also supporting the pkgconfig usage.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Closes #15

### CODE REVIEW

**Comment by jeremy-rifkin** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -172,65 +182,6 @@ outputs:
       doc_url: https://www.prevanders.net/libdwarfdoc/
       dev_url: https://github.com/davea42/libdwarf-code
 
-  - name: libdwarf-testing-meta
```
Comment (jeremy-rifkin) [inline recipe/meta.yaml:175]: I think this is no longer needed, however I would appreciate confirmation.

**Review by kkraus14 — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (kkraus14) [reply recipe/meta.yaml:175]: Agreed, thanks!

**Comment by kkraus14** [comment]
@conda-forge-admin, please rerender

**Comment by kkraus14** [comment]
@conda-forge-admin, please rerender

**Review by kkraus14 — APPROVED** [review]

**Comment by kkraus14** [comment]
Thanks @jeremy-rifkin!

**Comment by jeremy-rifkin** [comment]
Thank you so much!

**Pull request merged**